cdc/owner/changefeed.go

```diff
@@ -361,15 +364,17 @@ func (c *changefeed) tick(ctx cdcContext.Context, captures map[model.CaptureID]*
 		return nil
 	}
 
-	// If the owner is just initialized, barrierTs can be `checkpoint-1`.
+	// If the owner is just initialized, GlobalBarrierTs can be `tableCheckpoint-1`.
 	// In such case the `newResolvedTs` and `newCheckpointTs` may be larger
-	// than the barrierTs, but it shouldn't be, so we need to handle it here.
-	if newResolvedTs > barrierTs {
-		newResolvedTs = barrierTs
+	// than the GlobalBarrierTs, but it shouldn't be, so we need to handle it here.
+	if newResolvedTs > barrier.GlobalBarrierTs {
+		newResolvedTs = barrier.GlobalBarrierTs
 	}
-	if newCheckpointTs > barrierTs {
-		newCheckpointTs = barrierTs
+
+	if newCheckpointTs > minTableBarrierTs {
```

Comment:
Whether this situation is impossible to happen?

Reply:
When a changefeed is restarted, the minTableBarrierTs may regress to `checkpoint-1`.